lib/safe_pusher/client/rspec.rb

```diff
@@ -104,13 +109,28 @@ def branch
         `git rev-parse --abbrev-ref HEAD`.delete("\n")
       end
 
+      def create_new_file(path)
+        parent_directory = path.slice(0, path.rindex('/'))
+
+        FileUtils.mkdir_p(parent_directory) unless File.exist?(parent_directory)
+        File.open(path, 'w') {}
```

Comment:
To be slightly more readeable you could write it as:

```suggestion
        File.write(path, '')
```

Or, with `fileutils`

```suggestion
        FileUtils.touch(path)
```